src/main/java/vectorwing/farmersdelight/common/item/KnifeItem.java

```diff
@@ -81,7 +81,7 @@ public static void onCakeInteraction(PlayerInteractEvent.RightClickBlock event)
 			BlockState state = event.getWorld().getBlockState(pos);
 			Block block = state.getBlock();
 
-			if (BlockTags.CANDLE_CAKES.contains(block)) {
+			if (BlockTags.DROPS_CAKE_SLICE.contains(block)) {
```

Comment:
This line is pointing to the vanilla tags class, which doesn't contain your new mod tag.
You gotta change it to `ModTags.DROPS_CAKE_SLICE`.